R/qgis-result.R

```diff
@@ -49,6 +49,22 @@ qgis_error_output_does_not_exist <- function(x, which) {
   abort(glue("Result has no output '{ which }'.\nAvailable outputs are { available_outputs }"))
 }
 
+
+#' @keywords internal
+qgis_check_stdout <- function(x) {
+  if (qgis_result_status(x) == 0L && qgis_result_stdout(x) == "") {
+    warning(
```

Comment:
It was my attempt to follow your suggestion in https://github.com/paleolimbot/qgisprocess/issues/83#issuecomment-1029523135, regarding the `native:filedownloader` issue in Windows:

> Or maybe a solution is to warn if nothing comes through the JSON output? The problem might not be fixable from the QGIS perspective but if it's easy to detect the error then we should (i.e., check for `identical(result$stdout, "")`.

Please let me know what you prefer, considering the above.

Reply:
Well, I hadn't really thought much about warning vs error - no intuition involved :wink:. The thing with `native:filedownloader`, only in Windows (and first found by the automated check), was that it actually runs succesfully, hence status 0, but for some reason the standard output can't be captured in R with current approaches (even though stdout is printed in a Windows CMD console).

After reading a bit more into this (https://adv-r.hadley.nz/conditions.html), especially:

> Warnings, signalled by `warning()`, are weaker than errors: they signal that something has gone wrong, but the code has been able to recover and continue.

> There are only a couple of cases where using a warning is clearly appropriate:
>
> * When you __deprecate__ a function you want to allow older code to continue
  to work (so ignoring the warning is OK) but you want to encourage the user
  to switch to a new function.
>
> * When you are reasonably certain you can recover from a problem:
  If you were 100% certain that you could fix the problem, you wouldn't need 
  any message; if you were more uncertain that you could correctly fix the 
  issue, you'd throw an error.
>
> Otherwise use warnings with restraint, and carefully consider if an error would be more appropriate.

I think we should primarily consider the R-side in this. Even though algorithm X may have executed well, its output didn't come to R.

So my suggestion is to error as well! I'll propose an alternative.

Reply:
a1e695e